mountpoint-s3/src/cli.rs

```diff
@@ -812,54 +823,90 @@ where
     tracing::trace!("using metadata TTL setting {metadata_cache_ttl:?}");
     filesystem_config.cache_config = CacheConfig::new(metadata_cache_ttl);
 
-    if let Some(path) = &args.cache {
+    let create_disk_cache = |args: &CliArgs| -> anyhow::Result<_> {
+        assert!(args.max_cache_size.is_none() || args.max_cache_size.unwrap() > 0);
+        let path = args.cache.as_ref().expect("cache path must be some");
```

Comment:
Is it possible to change the function signature such that it can never be called with invalid args in the case that we check previously (which it seems we do)

Reply:
I can make `create_disk_cache` to receive unwrapped values instead of `&CliArgs`. The caller will need to unwrap them in some way, e.g:
```rust
create_disk_cache = |path: Path, max_cache_size: usize, cache_block_in_bytes: usize| -> anyhow::Result<_>
```